algosdk/v2client/algod.py

```diff
@@ -397,16 +397,58 @@ def genesis(self, **kwargs):
         req = "/genesis"
         return self.algod_request("GET", req, **kwargs)
 
-    def proof(self, round_num, txid, **kwargs):
+    def transaction_proof(
+        self, round_num, txid, hashtype="", response_format="json", **kwargs
+    ):
         """
         Get the proof for a given transaction in a round.
 
         Args:
             round_num (int): The round in which the transaction appears.
             txid (str): The transaction ID for which to generate a proof.
+            hashtype (str): The state proof hash type, sha256 or sha512_256.
+            response_format (str): the format in which the response is returned: either
+                "json" or "msgpack"
         """
+        params = {"format": response_format}
+        if hashtype != "":
+            params["hashtype"] = hashtype
         req = "/blocks/{}/transactions/{}/proof".format(round_num, txid)
-        return self.algod_request("GET", req, **kwargs)
+        return self.algod_request(
+            "GET",
+            req,
+            params=params,
+            response_format=response_format,
+            **kwargs
+        )
+
+    def lightheader(self, round_num, response_format="json", **kwargs):
```

Comment:
maybe we should rename it to `lightheader_proof`, `lightblockheader_proof` or something similar